GTBitsOfGood/fair-districts-ga
# Docker + Netlify

## Dockerize and Netlify

Issue Number(s): #87 #86 

Updates README to instruct how to use Docker to run the project. Also makes the .env variables come from Bitwarden and deploys the project to Netlify.

### Checklist

- [x] Production deployments must be migrated to Netlify (https://docs.netlify.com/site-deploys/create-deploys/)
- [x] Pull Request previews should also use Netlify (https://docs.netlify.com/site-deploys/deploy-previews/)
- [x] Pull Request Previews/Info is commented in the Github PR (https://docs.netlify.com/site-deploys/notifications/#github-pull-request-comments)
- [x] The netlify logo is added into the project frontend where you decide to be fit (this is necessary to adhere to our Netlify plan guidelines)
- [x] The reporisory has an Open Source License (Generally Apache or MIT are recommended)
- [x] One-command to spin up the development application in a containerized environment (docker-compose up)
- [x] Development application is accessible via host computer's browser (localhost:3000)
- [x] Local mysql is spun up in separate container accessible from default port
- [x] Application live-update is supported (through docker-compose/volumes)
- [x] Application can communicate with local mysql via Docker's inter-container networking (please message me if you need more info on this topic)
- [x] Add a section in the README explaining how to spin up a dev build with docker

### How to Test
Follow README instructions to deploy with Docker. Test that live-update works. Create commit and push to see if PR autocomment works. Visit preview site and test functionality.
